src/py/flwr/simulation/app.py

```diff
@@ -70,13 +72,29 @@
 
 """
 
+NodeToPartitionMapping = Dict[int, int]
+
+
+def _create_node_id_to_partition_mapping(
+    partition_ids: List[int],
+) -> NodeToPartitionMapping:
+    """Given a list of partition_ids, generate a node_id:partition_id mapping."""
+    nodes_mapping: NodeToPartitionMapping = {}  # {node-id; partition-id}
+    for i in partition_ids:
+        while True:
+            node_id = generate_rand_int_from_bytes(NODE_ID_NUM_BYTES)
+            if node_id not in nodes_mapping:
+                break
+        nodes_mapping[node_id] = i
+    return nodes_mapping
+
 
 # pylint: disable=too-many-arguments,too-many-statements,too-many-branches
 def start_simulation(
     *,
     client_fn: ClientFn,
     num_clients: Optional[int] = None,
-    clients_ids: Optional[List[str]] = None,
+    clients_ids: Optional[Union[List[int], List[str]]] = None,
```

Comment:
Should we add the `List[int]` variant as a separate argument `node_ids`? And deprecate `client_ids`?